src/resolve/resolved-bus.c

```diff
@@ -523,6 +524,9 @@ static int bus_method_resolve_hostname(sd_bus_message *message, void *userdata,
 
         q->bus_request = sd_bus_message_ref(message);
         q->request_family = family;
+        q->request_name = strdup(hostname);
+        if (!q->request_name)
+                return log_oom();
```

Comment:
A reference to `message` is kept, and `hostname` remains valid as long as `message` is alive. So there is no need to duplicate the string.

Reply:
Fixed in latest patch